jupyterhub/handlers/base.py

```diff
@@ -402,6 +414,7 @@ def finish_user_spawn(f=None):
                 spawner.add_poll_callback(self.user_stopped, user)
             finally:
                 spawner._proxy_pending = False
+                self.hub.spawn_pending_count -= 1
```

Comment:
The `hub` object is only meant to contain connection info (a tiny wrapper around URLs), not runtime info like this. Maybe store this in self.settings, instead?